app/forms/reset_password_form.rb

```diff
@@ -30,6 +30,8 @@ def valid_token
       # If the user is not saved in the database, that means looking them up by
       # their token failed
       errors.add(:reset_password_token, 'invalid_token', type: :invalid_token)
+    elsif invalid_account?
+      errors.add(:reset_password_token, 'token_expired', type: :token_expired)
     elsif !user.reset_password_period_valid?
       errors.add(:reset_password_token, 'token_expired', type: :token_expired)
```

Comment:
Since they use the same error, have them share this?
```suggestion
    elsif invalid_account? || !user.reset_password_period_valid?
      errors.add(:reset_password_token, 'token_expired', type: :token_expired)
```